design/overview.md

```diff
@@ -81,43 +82,36 @@ And provide a security equivalent to or greater than IBC.
+1. Each `Operator` configures their `Orchestrator` with the `Peggy Bridge ID` and `start threshold` they wish, their `Orchestrator` is to reject any proposals that do not agree on these values.
+1. Anyone submits a Peggy contract using a known codehash and the current validator set of the Cosmos zone to an Ethereum compatible blockchain. Note that step #1 must happen first because the `Orchestrator` Ethereum Addresses must be made known in order for verification to be possible.
+1. Each `Operator` may or may not configure their `Orchestrator` with the above Peggy contract address
+1. If configured with an address the `Orchestrator` checks the provided address. If the contract passes validation the `Orchestrator` signs and submits the contract address. Validation is defined as finding the expected code hash and a validator set matching the current set within `Allowed validator set delta`. This message also includes the proposed `Peggy Bridge ID` and `start threshold`
+1. A contract address is considered adopted when the proposed `start threshold`. Because validator sets change quickly `Orchestrators` not configured with a contract address observe the Cosmos blockchain for submissions. When an address is submitted they validate it and approve it themselves if it passes. This results in a workflow where once a valid contract is proposed it will be ratified in a matter of a few seconds.
+1. It is possible for the adoption process to fail if a race condition is intentionally created resulting in less than 66% of the validator power approving more than one valid Peggy Ethereum contract. In this case the Orchestrator will check the contract address with the majority of the power (or at random in the case of a perfect tie) and switch it's vote. This leaves only the possible edge case of >33% of `Operators` intentionally selecting a different contract address, bridge ID, or start thresholds. This would be a consensus failure and the bridge can not progress.
+1. The bridge ratification process is complete, the contract address is now placed in the store to be referenced only.
 
-1. Every *Operator* must initialize an *Orchestrator* on a secure computer
-1. Each *Operator* signs a `MsgRegisterOrchestrator` that ties the orchestrator's Cosmos and Ethereum keys to the validator (and makes them liable for slashing conditions)
-1. A `StartPeggy` message is triggered, either as a governance proposal or by any validator. If over X% (70? 80? 90?) of the validator power has registered, we create the original *MultiSig Set* and store that in the Cosmos chain.
+At this point, we know we have a contract on Ethereum with the proper _MultiSig Set_, that > `start threshold` of the _Orchestrator Set_ is online and agrees with this contract, and that the Cosmos chain has stored this contract address. Only then can we begin to accept transactions to transfer tokens
```

Comment:
Once we have this, do we want to send an "ActivateContract" message from Cosmos -> ETH to enable the contract.

Otherwise, proposed contracts that do not get accepted on the Cosmos Zone may already be accepting payments to transfer, but never be able to do so (leading to locked funds).

Sending an "ActivateBridge" message to the ethereum contract, like a "SendTokensToEth" message could be a nice step. If we don't want to add complexity to the ETH side, let's reuse existing messages. We can either define a "ValSetUpdate" message or "SendTokensToEth" with no tokens as a trigger to activate. Basically, the Ethereum contract must receive one valid message from the MultiSig set before it is activated and accepts ERC20 tokens.

Reply:
Well a dummy contract with no code can and will accept tokens. We would have to implement special code to send them back until the first valset update is made.

IMO 'Activate contract' is just checking that the validator set nonce > 0 because that means a chain has adopted the contract and started updating it.